extensions/tracing/src/main/java/io/opentelemetry/extension/tracing/Spanalyzer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.tracing;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import java.util.concurrent.Callable;
+
+/**
+ * Provides easy mechanisms for wrapping standard Java constructs with an OpenTelemetry Span.
+ *
+ * <p>Todo: figure out a better name for this.
+ */
+@SuppressWarnings("InconsistentOverloads")
+public class Spanalyzer {
```

Comment:
obviously, this is a ridiculous name. Suggestions extremely welcome.

Reply:
I know it was used in a different way in OpenTracing, but `SpanWrapper` kind of fits what you're doing here

Reply:
we also already have an internal class called `SpanWrapper` that could lead to confusion, but yes, this is one that I had considered.

Reply:
`ExtendedTracer` since presumably if moving these into core we'd add to `Tracer`?

Reply:
not bad. Or maybe `TracerExtension`?

Reply:
If kotlin, it'd definitely be `TracerExtensions` :) I guess per Java we'd want to name this what it is, which is an "extended tracer".